dotCMS/src/main/java/com/dotmarketing/portlets/cmsmaintenance/action/ViewCMSMaintenanceAction.java

```diff
@@ -479,7 +547,339 @@ private void moveAssetsToBackupDir() throws FileNotFoundException, IOException{
+					Mapper mapper = _xstream.getMapper();
+					_xstream.registerConverter(new HibernateCollectionConverter(mapper));
+					_xstream.registerConverter(new HibernateMapConverter(mapper));
+				}
+
+				/*
+				 * String _shortClassName =
+				 * clazz.getName().substring(clazz.getName().lastIndexOf("."),clazz.getName().length());
+				 * xstream.alias(_shortClassName, clazz);
+				 */
+				int i= 0;
+				int step = 1000;
+				int total =0;
+				java.text.NumberFormat formatter = new java.text.DecimalFormat("0000000000");
+				/* we will only export 10,000,000 items of any given type */
+				for(i=0;i < 10000000;i=i+step){
+
+					_dh = new HibernateUtil(clazz);
+					_dh.setFirstResult(i);
+					_dh.setMaxResults(step);
+
+					//This line was previously like;
+					//_dh.setQuery("from " + clazz.getName() + " order by 1,2");
+					//This caused a problem when the database is Oracle because Oracle causes problems when the results are ordered
+					//by an NCLOB field. In the case of dot_containers table, the second field, CODE, is an NCLOB field. Because of this,
+					//ordering is done only on the first field for the tables, which is INODE
+					if(com.dotmarketing.beans.Tree.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by parent, child, relation_type");
+					}
+					else if(MultiTree.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by parent1, parent2, child, relation_type");
+					}
+					else if(TagInode.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by inode, tag_id");
+					}
+					else if(Tag.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by tag_id, tagname");
+					}
+					else if(CalendarReminder.class.equals(clazz)){
+						_dh.setQuery("from " + clazz.getName() + " order by user_id, event_id, send_date");
+					}
+					else if(Identifier.class.equals(clazz)){
+						dc = new DotConnect();
+						dc.setSQL("select * from identifier order by parent_path, id")
+								.setStartRow(i).setMaxRows(step);
+					} else {
+						_dh.setQuery("from " + clazz.getName() + " order by 1");
+					}
+
+					if(Identifier.class.equals(clazz)){
+						_list = TransformerLocator
+								.createIdentifierTransformer(dc.loadObjectResults()).asList();
+					}else{
+						_list = _dh.list();
+					}
+
+					if(_list.size() ==0){
+						try {
+							_bout.close();
+						}
+						catch( java.lang.NullPointerException npe){}
+						_bout = null;
+
+						break;
+					}
+
+					if(_list != null && _list.size() > 0 && _list.get(0) instanceof Comparable){
+						java.util.Collections.sort(_list);
+					}
+
+					_writing = new File(backupTempFilePath + "/" + clazz.getName() + "_" + formatter.format(i) + ".xml");
+					_bout = new BufferedOutputStream(Files.newOutputStream(_writing.toPath()));
+
+					total = total + _list.size();
+
+					try {

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)